scripts/pre-commit.sh

```diff
@@ -65,6 +65,30 @@ function run-namespaced-e2e-tests() {
   return ${rc}
 }
 
+function check-kubebuilder-output() {
+  echo "Checking initial state of working tree"
+  if check-git-state; then
+    ./bin/kubebuilder generate
+    echo "Checking state of working tree after running 'kubebuilder generate'"
+    if check-git-state; then
+      exit 0
```

Comment:
We should not exit the script here. Simply returning a zero or non-zero return code should do the right thing since we have `errexit` set.